### ISSUE TITLE
Support for Built-in Preprocessing of Tensor Batches in YOLO Models

### ISSUE THREAD

**Issue by Renat2001** [ultralytics/ultralytics#16677]
### Search before asking

- [X] I have searched the Ultralytics [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar feature requests.

### Description

I am working with the Ultralytics YOLO model for inference, and I have noticed that the model currently supports built-in preprocessing for a list of NumPy arrays. However, it requires **manual preprocessing** when working with batches of PyTorch tensors. I would like to request support for built-in preprocessing of **tensor batches** similar to the functionality available for NumPy arrays.

### Use case

When working with a batch of image tensors (e.g., directly using ```torch.utils.data.DataLoader```), it would be helpful if the YOLO model could handle preprocessing internally. This would streamline the workflow and reduce the need for manual implementation of resizing, padding, and normalization steps.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @Renat2001, thank you for your interest in Ultralytics 🚀! We appreciate your suggestion regarding the support for built-in preprocessing of tensor batches in our YOLO models. 

This is an automated response to let you know that your feature request has been received. An Ultralytics engineer will respond to your request soon to discuss it further.

In the meantime, if this involves any 🐛 bug or specific implementation issues, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) if possible. This will help us in understanding the context better.

To explore more about using Ultralytics, please visit the [Docs](https://docs.ultralytics.com). You’ll find detailed [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) guides there.

You're also welcome to join the Ultralytics community where you can engage in discussions or seek help:
- For real-time chat, join us on [Discord](https://ultralytics.com/discord) 🎧.
- For detailed discussions, visit [Discourse](https://community.ultralytics.com).
- Connect with peers on our [Subreddit](https://reddit.com/r/ultralytics).

To ensure you're using the latest features and fixes, upgrade to the latest `ultralytics` package:

```bash
pip install -U ultralytics
```

You can run YOLO in various environments as listed below:
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM.
- **Amazon** Deep Learning AMI.
- **Docker Image**.

For the latest updates, check the CI status here:
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

We look forward to supporting your workflow with improved features!

**Comment by Y-T-G** [comment]
With `Tensors`, the expectation is that they are preprocessed. It's an option for people who want to perform their own preprocessing.

Why couldn't you just convert the `Tensors` to `numpy` if you want preprocessing to be performed? Because internally, it has to be turned into numpy anyway since the preprocessing uses OpenCV which can't work with `Tensors`.

**Comment by Renat2001** [comment]
> With `Tensors`, the expectation is that they are preprocessed. It's an option for people who want to perform their own preprocessing.
> 
> Why couldn't you just convert the `Tensors` to `numpy` if you want preprocessing to be performed? Because internally, it has to be turned into numpy anyway since the preprocessing uses OpenCV which can't work with `Tensors`.

Yes, you are right - I can convert ```Tensors``` to ```numpy``` what I do currently. But in the context of parallel processing with multi-GPU it would be redundant to place sampled batches from GPU devices to CPU for conversion and vice-versa to perform inference. 

On the other hand, I tried to apply manual preprocessing over ```Tensor``` like converting to ```float```, resizing and letterboxing. But model results significantly differ from results when I use ```numpy``` arrays.

**Comment by Y-T-G** [comment]
> But in the context of parallel processing with multi-GPU it would be redundant to place sampled batches from GPU devices to CPU for conversion and vice-versa to perform inference.

Yes. But it would have to be converted anyway because the preprocessing function used by `ultralytics` can't work with GPU tensors.

**Comment by Renat2001** [comment]
> > But in the context of parallel processing with multi-GPU it would be redundant to place sampled batches from GPU devices to CPU for conversion and vice-versa to perform inference.
> 
> Yes. But it would have to be converted anyway because the preprocessing function used by `ultralytics` can't work with GPU tensors.

Ok. But have you tried to imitate the ```ultralytics``` prepocessing step with ```Tensor``` operations? Or there is no way to do it?

**Comment by Y-T-G** [comment]
You could try pasting the preprocessing from here into GPT and asking it to create an equivalent for tensors.

https://github.com/ultralytics/ultralytics/blob/fceaa4003871479e047f607ebb09356a95475e33/examples/YOLOv8-OpenCV-int8-tflite-Python/main.py
